epochStart/shardchain/trigger.go

```diff
@@ -470,12 +508,20 @@ func (t *trigger) getHeaderWithNonceAndPrevHashFromCache(nonce uint64, prevHash
 			continue
 		}
 
+		if lowestRound > hdrWithNonce.GetRound() {
+			lowestRound = hdrWithNonce.GetRound()
+			chosenMeta = hdrWithNonce
+		}
+
 		t.mapHashHdr[string(hashes[i])] = hdrWithNonce
 		t.mapNonceHashes[hdrWithNonce.Nonce] = append(t.mapNonceHashes[hdrWithNonce.Nonce], string(hashes[i]))
-		return hdrWithNonce
 	}
 
-	return nil
+	if chosenMeta.GetNonce() != nonce {
```

Comment:
The same comment here, you can use  if lowestRound == uint64(math.MaxUint64) instead this one

Reply:
no way to call this with nonce 0

Reply:
I know, but with this check you will cover any situation